python/deltalake/table.py

```diff
@@ -258,6 +259,15 @@ def __init__(
             log_buffer_size=log_buffer_size,
         )
         self._metadata = Metadata(self._table)
+        
+        if lazy_load:
+            self._table = RawDeltaTable.load_lazy(table_uri)
+            self._metadata = None
+            return
+        self._table = RawDeltaTable(
+            table_uri, version=version, storage_options=storage_options
+        )
+        self._metadata = Metadata(self._table)
```

Comment:
I think it would be cleaner if you do if-else here:

```suggestion
        if lazy_load:
            self._table = RawDeltaTable.load_lazy(table_uri)
            self._metadata = None
        else:
            self._table = RawDeltaTable(
                    table_uri, version=version, storage_options=storage_options
            )
            self._metadata = Metadata(self._table)
```